Orleans.Providers.MongoDB/ServiceCollectionExtensions.cs

```diff
@@ -16,5 +21,25 @@ public static IServiceCollection AddMongoDBClient(this IServiceCollection servic
 
             return services;
         }
+
+        /// <summary>
+        /// Configure silo to use MongoDb with a passed in connection string.
+        /// </summary>
+        /// <param name="services">The services.</param>
+        /// <param name="configureClientSettings">The configuration delegate to configure the <see cref="MongoClientSettings"/>.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddMongoDBClient(this IServiceCollection services, ActionRef<MongoClientSettings> configureClientSettings)
```

Comment:
What are your thoughts on using `Func<IServiceProvider, MongoClientSettings> settingsFactory` instead of the `Action<MongoClientSettings>`?
This will allow developers to access the service provider in order to configure the settings, for example, you can do something like:

```csharp
.UseMongoDBClient(sp =>
{
    var options = sp.GetRequiredService<IOptions<MongoDBConfiguration>>();
    var settings = MongoClientSettings.FromConnectionString(options.Value.ConnectionString);
    settings.Credential = MongoCredential.CreateCredential(options.Value.Database, options.Value.User, options.Value.Password);
    return settings;
})
```

Reply:
Also the other overload can be refactored into:
```csharp
public static IServiceCollection AddMongoDBClient(this IServiceCollection services, string connectionString)
{
    return services.AddMongoDBClient(sp => MongoClientSettings.FromConnectionString(connectionString));
}
```

Reply:
> What are your thoughts on using `Func<IServiceProvider, MongoClientSettings> settingsFactory` instead of the `Action<MongoClientSettings>`?
I like it a lot! Looks much cleaner. See latest push.



> Also the other overload can be refactored into:
> 
> ```cs
> public static IServiceCollection AddMongoDBClient(this IServiceCollection services, string connectionString)
> {
>     return services.AddMongoDBClient(sp => MongoClientSettings.FromConnectionString(connectionString));
> }
> ```

Also done.